### ISSUE TITLE
--api and --query tags do not generate all the files

### ISSUE THREAD

**Issue by sriv** [lsegal/yard#836]
I have a few methods in my gem that are supposed to be used by consumers. I marked them as:

`@api public`

When I use this to filter out docs, I hit a dead end.

When I do `yard doc`, I see all the files generated, the docs look awesome.

When I do wither `yard doc --api public` or `yard doc --query '@api.text == "public"'`, it generates the `frames.html`, but does not list the Classes right. The methods are listed, but method specific files are not generated.

Here is the debug trace:

```
steam$ yard  --api public --debug
[debug]: Parsing ["{lib,app}/**/*.rb", "ext/**/*.c"] with `ruby` parser
[debug]: Parsing lib/table.rb
[debug]: Parsing lib/gauge.rb
[debug]: Parsing lib/api.pb.rb
[debug]: Parsing lib/spec.pb.rb
[debug]: Parsing lib/executor.rb
[debug]: Parsing lib/datastore.rb
[debug]: Parsing lib/connector.rb
[debug]: Parsing lib/messages.pb.rb
[debug]: Parsing lib/code_parser.rb
[debug]: Parsing lib/method_cache.rb
[debug]: Parsing lib/configuration.rb
[debug]: Parsing lib/gauge_runtime.rb
[debug]: Parsing lib/message_processor.rb
[debug]: Parsing lib/processors/execution_handler.rb
[debug]: Parsing lib/processors/kill_request_processor.rb
[debug]: Parsing lib/processors/datastore_init_processor.rb
[debug]: Parsing lib/processors/execution_hook_processors.rb
[debug]: Parsing lib/processors/step_name_request_processor.rb
[debug]: Parsing lib/processors/step_names_request_processor.rb
[debug]: Parsing lib/processors/execute_step_request_processor.rb
[debug]: Parsing lib/processors/refactor_step_request_processor.rb
[debug]: Parsing lib/processors/step_validation_request_processor.rb
[debug]: Serializing to .yardoc/objects/root.dat
[debug]: Generating asset js/jquery.js
[debug]: Serializing to doc/js/jquery.js
[debug]: Generating asset js/app.js
[debug]: Serializing to doc/js/app.js
[debug]: Generating asset js/full_list.js
[debug]: Serializing to doc/js/full_list.js
[debug]: Generating asset css/style.css
[debug]: Serializing to doc/css/style.css
[debug]: Generating asset css/common.css
[debug]: Serializing to doc/css/common.css
[debug]: Generating asset css/full_list.css
[debug]: Serializing to doc/css/full_list.css
[debug]: Generating asset class_list.html
[debug]: Serializing to doc/class_list.html
[debug]: Generating asset method_list.html
[debug]: Serializing to doc/method_list.html
[debug]: Generating asset file_list.html
[debug]: Serializing to doc/file_list.html
[debug]: Generating asset frames.html
[debug]: Serializing to doc/frames.html
[debug]: Serializing to doc/_index.html
[debug]: Serializing to doc/index.html
[debug]: Serializing to doc/file.README.html
Files:           4
Modules:         0 (    0 undocumented)
Classes:         0 (    0 undocumented)
Constants:       0 (    0 undocumented)
Methods:        17 (    0 undocumented)
 100.00% documented
```

The debug output from `yard --debug` looks like this:
```
yard  --debug
[debug]: Parsing ["{lib,app}/**/*.rb", "ext/**/*.c"] with `ruby` parser
[debug]: Parsing lib/table.rb
[debug]: Parsing lib/gauge.rb
[debug]: Parsing lib/api.pb.rb
[debug]: Parsing lib/spec.pb.rb
[debug]: Parsing lib/executor.rb
[debug]: Parsing lib/datastore.rb
[debug]: Parsing lib/connector.rb
[debug]: Parsing lib/messages.pb.rb
[debug]: Parsing lib/code_parser.rb
[debug]: Parsing lib/method_cache.rb
[debug]: Parsing lib/configuration.rb
[debug]: Parsing lib/gauge_runtime.rb
[debug]: Parsing lib/message_processor.rb
[debug]: Parsing lib/processors/execution_handler.rb
[debug]: Parsing lib/processors/kill_request_processor.rb
[debug]: Parsing lib/processors/datastore_init_processor.rb
[debug]: Parsing lib/processors/execution_hook_processors.rb
[debug]: Parsing lib/processors/step_name_request_processor.rb
[debug]: Parsing lib/processors/step_names_request_processor.rb
[debug]: Parsing lib/processors/execute_step_request_processor.rb
[debug]: Parsing lib/processors/refactor_step_request_processor.rb
[debug]: Parsing lib/processors/step_validation_request_processor.rb
[debug]: Serializing to .yardoc/objects/root.dat
[debug]: Re-generating object ...
[debug]: Re-generating object Gauge...
[debug]: Re-generating object Kernel...
[debug]: Re-generating object Gauge::Messages...
[debug]: Re-generating object Gauge::Messages::APIMessage::APIMessageType...
[debug]: Re-generating object Gauge::Messages::ProtoItem::ItemType...
[debug]: Re-generating object Gauge::Messages::Fragment::FragmentType...
[debug]: Re-generating object Gauge::Messages::Parameter::ParameterType...
[debug]: Re-generating object Gauge::Executor...
[debug]: Re-generating object Gauge::Connector...
[debug]: Re-generating object Gauge::Messages::Message::MessageType...
[debug]: Re-generating object Gauge::Runtime...
[debug]: Re-generating object Gauge::Processors...
[debug]: Re-generating object Gauge::Processors::ExecutionHandler...
[debug]: Re-generating object Gauge::Table...
[debug]: Re-generating object Gauge::Messages::GetProjectRootRequest...
[debug]: Re-generating object Gauge::Messages::GetProjectRootResponse...
[debug]: Re-generating object Gauge::Messages::GetInstallationRootRequest...
[debug]: Re-generating object Gauge::Messages::GetInstallationRootResponse...
[debug]: Re-generating object Gauge::Messages::GetAllStepsRequest...
[debug]: Re-generating object Gauge::Messages::GetAllStepsResponse...
[debug]: Re-generating object Gauge::Messages::GetAllSpecsRequest...
[debug]: Re-generating object Gauge::Messages::GetAllSpecsResponse...
[debug]: Re-generating object Gauge::Messages::GetAllConceptsRequest...
[debug]: Re-generating object Gauge::Messages::GetAllConceptsResponse...
[debug]: Re-generating object Gauge::Messages::ConceptInfo...
[debug]: Re-generating object Gauge::Messages::GetStepValueRequest...
[debug]: Re-generating object Gauge::Messages::GetStepValueResponse...
[debug]: Re-generating object Gauge::Messages::GetLanguagePluginLibPathRequest...
[debug]: Re-generating object Gauge::Messages::GetLanguagePluginLibPathResponse...
[debug]: Re-generating object Gauge::Messages::ErrorResponse...
[debug]: Re-generating object Gauge::Messages::PerformRefactoringRequest...
[debug]: Re-generating object Gauge::Messages::PerformRefactoringResponse...
[debug]: Re-generating object Gauge::Messages::APIMessage...
[debug]: Re-generating object Gauge::Messages::ProtoSpec...
[debug]: Re-generating object Gauge::Messages::ProtoItem...
[debug]: Re-generating object Gauge::Messages::ProtoScenario...
[debug]: Re-generating object Gauge::Messages::ProtoTableDrivenScenario...
[debug]: Re-generating object Gauge::Messages::ProtoStep...
[debug]: Re-generating object Gauge::Messages::ProtoConcept...
[debug]: Re-generating object Gauge::Messages::ProtoTags...
[debug]: Re-generating object Gauge::Messages::Fragment...
[debug]: Re-generating object Gauge::Messages::Parameter...
[debug]: Re-generating object Gauge::Messages::ProtoComment...
[debug]: Re-generating object Gauge::Messages::ProtoTable...
[debug]: Re-generating object Gauge::Messages::ProtoTableRow...
[debug]: Re-generating object Gauge::Messages::ProtoStepExecutionResult...
[debug]: Re-generating object Gauge::Messages::ProtoExecutionResult...
[debug]: Re-generating object Gauge::Messages::ProtoHookFailure...
[debug]: Re-generating object Gauge::Messages::ProtoSuiteResult...
[debug]: Re-generating object Gauge::Messages::ProtoSpecResult...
[debug]: Re-generating object Gauge::Messages::ProtoStepValue...
[debug]: Re-generating object Gauge::DataStore...
[debug]: Re-generating object Gauge::DataStoreFactory...
[debug]: Re-generating object Gauge::Messages::KillProcessRequest...
[debug]: Re-generating object Gauge::Messages::ExecutionStatusResponse...
[debug]: Re-generating object Gauge::Messages::ExecutionStartingRequest...
[debug]: Re-generating object Gauge::Messages::SpecExecutionStartingRequest...
[debug]: Re-generating object Gauge::Messages::SpecExecutionEndingRequest...
[debug]: Re-generating object Gauge::Messages::ScenarioExecutionStartingRequest...
[debug]: Re-generating object Gauge::Messages::ScenarioExecutionEndingRequest...
[debug]: Re-generating object Gauge::Messages::StepExecutionStartingRequest...
[debug]: Re-generating object Gauge::Messages::StepExecutionEndingRequest...
[debug]: Re-generating object Gauge::Messages::ExecutionInfo...
[debug]: Re-generating object Gauge::Messages::SpecInfo...
[debug]: Re-generating object Gauge::Messages::ScenarioInfo...
[debug]: Re-generating object Gauge::Messages::StepInfo...
[debug]: Re-generating object Gauge::Messages::ExecuteStepRequest...
[debug]: Re-generating object Gauge::Messages::StepValidateRequest...
[debug]: Re-generating object Gauge::Messages::StepValidateResponse...
[debug]: Re-generating object Gauge::Messages::ExecutionEndingRequest...
[debug]: Re-generating object Gauge::Messages::SuiteExecutionResult...
[debug]: Re-generating object Gauge::Messages::StepNamesRequest...
[debug]: Re-generating object Gauge::Messages::StepNamesResponse...
[debug]: Re-generating object Gauge::Messages::ScenarioDataStoreInitRequest...
[debug]: Re-generating object Gauge::Messages::SpecDataStoreInitRequest...
[debug]: Re-generating object Gauge::Messages::SuiteDataStoreInitRequest...
[debug]: Re-generating object Gauge::Messages::ParameterPosition...
[debug]: Re-generating object Gauge::Messages::RefactorRequest...
[debug]: Re-generating object Gauge::Messages::RefactorResponse...
[debug]: Re-generating object Gauge::Messages::StepNameRequest...
[debug]: Re-generating object Gauge::Messages::StepNameResponse...
[debug]: Re-generating object Gauge::Messages::Message...
[debug]: Re-generating object Gauge::CodeParser...
[debug]: Re-generating object Gauge::MethodCache...
[debug]: Re-generating object Gauge::Configuration...
[debug]: Re-generating object Gauge::MessageProcessor...
[debug]: Generating asset js/jquery.js
[debug]: Serializing to doc/js/jquery.js
[debug]: Generating asset js/app.js
[debug]: Serializing to doc/js/app.js
[debug]: Generating asset js/full_list.js
[debug]: Serializing to doc/js/full_list.js
[debug]: Generating asset css/style.css
[debug]: Serializing to doc/css/style.css
[debug]: Generating asset css/common.css
[debug]: Serializing to doc/css/common.css
[debug]: Generating asset css/full_list.css
[debug]: Serializing to doc/css/full_list.css
[debug]: Generating asset class_list.html
[debug]: Serializing to doc/class_list.html
[debug]: Generating asset method_list.html
[debug]: Serializing to doc/method_list.html
[debug]: Generating asset file_list.html
[debug]: Serializing to doc/file_list.html
[debug]: Generating asset frames.html
[debug]: Serializing to doc/frames.html
[debug]: Serializing to doc/_index.html
[debug]: Serializing to doc/index.html
[debug]: Serializing to doc/file.README.html
[debug]: Serializing to doc/top-level-namespace.html
[debug]: Serializing to doc/Gauge.html
[debug]: Serializing to doc/Kernel.html
[debug]: Serializing to doc/Gauge/Messages.html
[debug]: Serializing to doc/Gauge/Messages/APIMessage/APIMessageType.html
[debug]: Serializing to doc/Gauge/Messages/ProtoItem/ItemType.html
[debug]: Serializing to doc/Gauge/Messages/Fragment/FragmentType.html
[debug]: Serializing to doc/Gauge/Messages/Parameter/ParameterType.html
[debug]: Serializing to doc/Gauge/Executor.html
[debug]: Serializing to doc/Gauge/Connector.html
[debug]: Serializing to doc/Gauge/Messages/Message/MessageType.html
[debug]: Serializing to doc/Gauge/Runtime.html
[debug]: Serializing to doc/Gauge/Processors.html
[debug]: Serializing to doc/Gauge/Processors/ExecutionHandler.html
[debug]: Serializing to doc/Gauge/Table.html
[debug]: Serializing to doc/Gauge/Messages/GetProjectRootRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetProjectRootResponse.html
[debug]: Serializing to doc/Gauge/Messages/GetInstallationRootRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetInstallationRootResponse.html
[debug]: Serializing to doc/Gauge/Messages/GetAllStepsRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetAllStepsResponse.html
[debug]: Serializing to doc/Gauge/Messages/GetAllSpecsRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetAllSpecsResponse.html
[debug]: Serializing to doc/Gauge/Messages/GetAllConceptsRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetAllConceptsResponse.html
[debug]: Serializing to doc/Gauge/Messages/ConceptInfo.html
[debug]: Serializing to doc/Gauge/Messages/GetStepValueRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetStepValueResponse.html
[debug]: Serializing to doc/Gauge/Messages/GetLanguagePluginLibPathRequest.html
[debug]: Serializing to doc/Gauge/Messages/GetLanguagePluginLibPathResponse.html
[debug]: Serializing to doc/Gauge/Messages/ErrorResponse.html
[debug]: Serializing to doc/Gauge/Messages/PerformRefactoringRequest.html
[debug]: Serializing to doc/Gauge/Messages/PerformRefactoringResponse.html
[debug]: Serializing to doc/Gauge/Messages/APIMessage.html
[debug]: Serializing to doc/Gauge/Messages/ProtoSpec.html
[debug]: Serializing to doc/Gauge/Messages/ProtoItem.html
[debug]: Serializing to doc/Gauge/Messages/ProtoScenario.html
[debug]: Serializing to doc/Gauge/Messages/ProtoTableDrivenScenario.html
[debug]: Serializing to doc/Gauge/Messages/ProtoStep.html
[debug]: Serializing to doc/Gauge/Messages/ProtoConcept.html
[debug]: Serializing to doc/Gauge/Messages/ProtoTags.html
[debug]: Serializing to doc/Gauge/Messages/Fragment.html
[debug]: Serializing to doc/Gauge/Messages/Parameter.html
[debug]: Serializing to doc/Gauge/Messages/ProtoComment.html
[debug]: Serializing to doc/Gauge/Messages/ProtoTable.html
[debug]: Serializing to doc/Gauge/Messages/ProtoTableRow.html
[debug]: Serializing to doc/Gauge/Messages/ProtoStepExecutionResult.html
[debug]: Serializing to doc/Gauge/Messages/ProtoExecutionResult.html
[debug]: Serializing to doc/Gauge/Messages/ProtoHookFailure.html
[debug]: Serializing to doc/Gauge/Messages/ProtoSuiteResult.html
[debug]: Serializing to doc/Gauge/Messages/ProtoSpecResult.html
[debug]: Serializing to doc/Gauge/Messages/ProtoStepValue.html
[debug]: Serializing to doc/Gauge/DataStore.html
[debug]: Serializing to doc/Gauge/DataStoreFactory.html
[debug]: Serializing to doc/Gauge/Messages/KillProcessRequest.html
[debug]: Serializing to doc/Gauge/Messages/ExecutionStatusResponse.html
[debug]: Serializing to doc/Gauge/Messages/ExecutionStartingRequest.html
[debug]: Serializing to doc/Gauge/Messages/SpecExecutionStartingRequest.html
[debug]: Serializing to doc/Gauge/Messages/SpecExecutionEndingRequest.html
[debug]: Serializing to doc/Gauge/Messages/ScenarioExecutionStartingRequest.html
[debug]: Serializing to doc/Gauge/Messages/ScenarioExecutionEndingRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepExecutionStartingRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepExecutionEndingRequest.html
[debug]: Serializing to doc/Gauge/Messages/ExecutionInfo.html
[debug]: Serializing to doc/Gauge/Messages/SpecInfo.html
[debug]: Serializing to doc/Gauge/Messages/ScenarioInfo.html
[debug]: Serializing to doc/Gauge/Messages/StepInfo.html
[debug]: Serializing to doc/Gauge/Messages/ExecuteStepRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepValidateRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepValidateResponse.html
[debug]: Serializing to doc/Gauge/Messages/ExecutionEndingRequest.html
[debug]: Serializing to doc/Gauge/Messages/SuiteExecutionResult.html
[debug]: Serializing to doc/Gauge/Messages/StepNamesRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepNamesResponse.html
[debug]: Serializing to doc/Gauge/Messages/ScenarioDataStoreInitRequest.html
[debug]: Serializing to doc/Gauge/Messages/SpecDataStoreInitRequest.html
[debug]: Serializing to doc/Gauge/Messages/SuiteDataStoreInitRequest.html
[debug]: Serializing to doc/Gauge/Messages/ParameterPosition.html
[debug]: Serializing to doc/Gauge/Messages/RefactorRequest.html
[debug]: Serializing to doc/Gauge/Messages/RefactorResponse.html
[debug]: Serializing to doc/Gauge/Messages/StepNameRequest.html
[debug]: Serializing to doc/Gauge/Messages/StepNameResponse.html
[debug]: Serializing to doc/Gauge/Messages/Message.html
[debug]: Serializing to doc/Gauge/CodeParser.html
[debug]: Serializing to doc/Gauge/MethodCache.html
[debug]: Serializing to doc/Gauge/Configuration.html
[debug]: Serializing to doc/Gauge/MessageProcessor.html
Files:          23
Modules:        13 (    7 undocumented)
Classes:        73 (   69 undocumented)
Constants:      57 (   57 undocumented)
Methods:        73 (   51 undocumented)
 14.81% documented
```

I notice that the run with no filter generates one file per class/module, which does not happen with the `--api` filter.

I am on `ruby 2.2`, `yard 0.8.7.6`.

**Comment by lsegal** [comment]
Can you provide a minimal reproduction of what you are doing and what the expectation is? If you are only adding `@api` tags on the individual methods themselves, then this is expected behavior-- you are filtering out all objects that do not have `@api public`, which includes your classes/modules.

tl;dr you likely want to be adding `@api public` to your modules/classes, not individual methods. See the [@api docs](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#api) for notes in addition to the important note of `@api` being transitive by default, meaning applying `@api public` to your class immediately applies this tag to all inner-objects. If you use this pattern, you apply the tag to the class and then apply `@api private` to individual methods when you want to *opt-out* of public behavior.

**Comment by sriv** [comment]
Sorry for the dump. You're right, I was adding the `@api` tag at method level, without noticing the transitive nature of it. I will restructure my tags and check it out. Thanks.
